device_backends/LogicalNameMapping/src/LNMDoubleBufferPlugin.cc

```diff
@@ -0,0 +1,124 @@
+#include "LNMDoubleBufferPlugin.h"
+
+#include "BackendFactory.h"
+
+std::string getValue(const std::map<std::string, std::string>& m, std::string key) {
+  auto it = m.find(key);
+  if(it != m.end()) {
+    return it->second;
+  }
+  else {
+    std::string message =
+        "LogicalNameMappingBackend DoubleBufferPlugin: Missing parameter " + std::string("'") + key + "'.";
+    throw ChimeraTK::logic_error(message);
+  }
+}
+
+namespace ChimeraTK { namespace LNMBackend {
+  DoubleBufferPlugin::DoubleBufferPlugin(LNMBackendRegisterInfo info, std::map<std::string, std::string> parameters)
+  : AccessorPlugin(info), _parameters(std::move(parameters)) {
+    _targetDeviceName = info.deviceName;
+  }
+
+  void DoubleBufferPlugin::updateRegisterInfo(BackendRegisterCatalogue<LNMBackendRegisterInfo>& catalogue) {
+    // first update the info so we have the latest version from the catalogue.
+    _info = catalogue.getBackendRegister(_info.name);
+    // Change register info to read-only
+    _info.writeable = false;
+    catalogue.modifyRegister(_info);
+  }
+
+  /*************************************************/
+  /** Helper class to implement MultiplierPlugin::decorateAccessor (can later be realised with if constexpr) */
+  template<typename UserType, typename TargetType>
+  struct DoubleBufferPlugin_Helper {
+    static boost::shared_ptr<NDRegisterAccessor<UserType>> decorateAccessor(
+        boost::shared_ptr<LogicalNameMappingBackend>& /*backend*/, boost::shared_ptr<NDRegisterAccessor<TargetType>>&,
+        const std::map<std::string, std::string>&, std::string) {
+      assert(false); // only specialisation is valid
+      return {};
+    }
+  };
+  template<typename UserType>
+  struct DoubleBufferPlugin_Helper<UserType, UserType> {
+    static boost::shared_ptr<NDRegisterAccessor<UserType>> decorateAccessor(
+        boost::shared_ptr<LogicalNameMappingBackend>& backend, boost::shared_ptr<NDRegisterAccessor<UserType>>& target,
+        const std::map<std::string, std::string>& parameters, std::string targetDeviceName) {
+      return boost::make_shared<DoubleBufferAccessorDecorator<UserType>>(backend, target, parameters, targetDeviceName);
+    }
+  };
+  /*************************************************/
+  template<typename UserType, typename TargetType>
+  boost::shared_ptr<NDRegisterAccessor<UserType>> DoubleBufferPlugin::decorateAccessor(
+      boost::shared_ptr<LogicalNameMappingBackend>& backend,
+      boost::shared_ptr<NDRegisterAccessor<TargetType>>& target) const {
+    return DoubleBufferPlugin_Helper<UserType, TargetType>::decorateAccessor(
```

Comment:
This can be simplified with modern C++. No `DoubleBufferPlugin_Helper` helper is required. Just use
```
if constexpr(std::is_same<UserType,TargetType>::value) {
  return boost::make_shared<DoubleBufferAccessorDecorator<UserType>>(backend, target, parameters, targetDeviceName);
}
else {
  assert(false);
}
```

Reply:
Very nice hint. 
I redefined the 
`template<typename UserType, typename TargetType>
  boost::shared_ptr<NDRegisterAccessor<UserType>> DoubleBufferPlugin::decorateAccessor(
      boost::shared_ptr<LogicalNameMappingBackend>& backend,
      boost::shared_ptr<NDRegisterAccessor<TargetType>>& target) const  {....}`
according to your idea. `DoubleBufferPlugin_Helper ` is not needed.

Just a side note. The same plugin mechanism is used for other plugins - `ForcePollingReadPlugin_Helper` , `ForceReadOnlyPlugin_Helper` etc. Is is a good oportunity to change it also there or should it be done separatelly ?

Reply:
I created a ticket: https://redmine.msktools.desy.de/issues/9741